dkeycache/App/ra_client.cpp

```diff
@@ -197,6 +203,32 @@ static int RetreiveDomainKey(int32_t sockfd) {
         goto CLEANUP;
     }
 
+    // call initsession and get sessionid
+    // stroe sessionid to g_sessionid
+    // msg1\msg3 will add sessionid in their header
+    p_sessionId_req = (ra_samp_request_header_t*)
+                     malloc(sizeof(ra_samp_request_header_t));
+    if(NULL == p_sessionId_req) {
+        ret = -1;
+        goto CLEANUP;
+    }
+    p_sessionId_req->type = TYPE_RA_GET_SESSION_ID_REQ;
+    p_sessionId_req->size = 0;
+    SendAndRecvMsg(sockfd, p_sessionId_req, &p_sessionId_res);
+    if(!p_sessionId_res) {
+        printf("Error, get session id failed.\n");
+        ret = -1;
+        goto CLEANUP;
+    }
+    if(TYPE_RA_GET_SESSION_ID_RES != p_sessionId_res->type) {
+        printf("Error, Session id response type is not matched!\n");
+        ret = -1;
+        goto CLEANUP;
+    }
+    printf("session id recieved success!\n");
+    sessionId = (uint8_t *)calloc(SESSION_ID_SIZE, sizeof(uint8_t));
```

Comment:
seesionId not free? and suggest to use sessionID[SESSION_ID_SIZE] as instead since the size is fixed

Reply:
use sessionID[SESSION_ID_SIZE]，and test passed